### ISSUE TITLE
[a11y]: Tag close button isn't properly labeled

### ISSUE THREAD

**Issue by finken2** [carbon-design-system/carbon#14607]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.37

### React version

17

### Automated testing tool and ruleset

Accessibiility checker / look at the dom

### Description

The fix for [this issue](https://github.com/carbon-design-system/carbon/issues/13706)  (PR [here](https://github.com/carbon-design-system/carbon/pull/13852)) isn't quite right. 

The close button of the tag has `aria-labelledby` pointing to the tag, but the tag doesn't have a body that makes sense for `aria-labelledby` of the close button. From the storybook:
<img width="816" alt="image" src="https://github.com/carbon-design-system/carbon/assets/50754824/b5a4db01-c155-445b-8ad6-67d9917030c5">

The close button should have an aria label that says "close tag" or something along those lines. We already have the `title` prop that looks to be the same thing, so maybe just change this `aria-labelledby={tagId}` ---> `aria-label={title}`

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tag--playground&args=filter:!true

### Steps to reproduce

storybook, Tag, turn on filter, look at the dom as mentioned above.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
